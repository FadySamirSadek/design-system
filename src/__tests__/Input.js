import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { Input } from '..'

describe('Input', () => {
  test('it renders', () => {
    const json = renderer.create(<Input id="renders" />).toJSON()
    expect(json).toMatchSnapshot()
  })
  test('it renders an input element with a red border with a color prop is passed', () => {
    const json = renderer.create(<Input color="red" id="color" />).toJSON()
    expect(json).toMatchSnapshot()
  })
  test('it renders an input element with a really large padding and margin', () => {
    const json = renderer.create(<Input p={4} m={4} id="space" />).toJSON()
    expect(json).toMatchSnapshot()
  })
  test('it renders an input element with large text', () => {
    const json = renderer.create(<Input fontSize={4} id="fontSize" />).toJSON()
    expect(json).toMatchSnapshot()
  })
})
