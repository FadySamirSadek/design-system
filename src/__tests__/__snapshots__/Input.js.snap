// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders 1`] = `
.c0 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: block;
  vertical-align: middle;
  max-width: 32rem;
  min-height: 36px;
  line-height: inherit;
  font-family: inherit;
  background-color: transparent;
  border-radius: 4px;
  border-width: 1px;
  border-style: solid;
  border-color: #dde1e4;
  -webkit-transition: 0.125s box-shadow ease-out;
  transition: 0.125s box-shadow ease-out;
  font-size: 16px;
  margin: 0px;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  padding-bottom: 4px;
  width: 100%;
  color: inherit;
  background-color: transparent;
}

.c0::-ms-clear {
  display: none;
}

.c0::-ms-clear {
  display: none;
}

.c0::-ms-clear {
  display: none;
}

.c0::-ms-clear {
  display: none;
}

.c0:focus {
  outline: none;
  border-color: #2d9ee4;
  box-shadow: 0 0 0 2px #add9f5;
}

<input
  className="c0"
  color="inherit"
  fontSize={2}
  id="renders"
/>
`;

exports[`Input renders with custom fontSize 1`] = `
.c0 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: block;
  vertical-align: middle;
  max-width: 32rem;
  min-height: 36px;
  line-height: inherit;
  font-family: inherit;
  background-color: transparent;
  border-radius: 4px;
  border-width: 1px;
  border-style: solid;
  border-color: #dde1e4;
  -webkit-transition: 0.125s box-shadow ease-out;
  transition: 0.125s box-shadow ease-out;
  font-size: 24px;
  margin: 0px;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  padding-bottom: 4px;
  width: 100%;
  color: inherit;
  background-color: transparent;
}

.c0::-ms-clear {
  display: none;
}

.c0::-ms-clear {
  display: none;
}

.c0::-ms-clear {
  display: none;
}

.c0::-ms-clear {
  display: none;
}

.c0:focus {
  outline: none;
  border-color: #2d9ee4;
  box-shadow: 0 0 0 2px #add9f5;
}

<input
  className="c0"
  color="inherit"
  fontSize={4}
  id="fontSize"
/>
`;

exports[`Input renders with custom spacing 1`] = `
.c0 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: block;
  vertical-align: middle;
  max-width: 32rem;
  min-height: 36px;
  line-height: inherit;
  font-family: inherit;
  background-color: transparent;
  border-radius: 4px;
  border-width: 1px;
  border-style: solid;
  border-color: #dde1e4;
  -webkit-transition: 0.125s box-shadow ease-out;
  transition: 0.125s box-shadow ease-out;
  font-size: 16px;
  margin: 32px;
  padding: 32px;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  padding-bottom: 4px;
  width: 100%;
  color: inherit;
  background-color: transparent;
}

.c0::-ms-clear {
  display: none;
}

.c0::-ms-clear {
  display: none;
}

.c0::-ms-clear {
  display: none;
}

.c0::-ms-clear {
  display: none;
}

.c0:focus {
  outline: none;
  border-color: #2d9ee4;
  box-shadow: 0 0 0 2px #add9f5;
}

<input
  className="c0"
  color="inherit"
  fontSize={2}
  id="space"
/>
`;
